sklearn/tests/test_discriminant_analysis.py

```diff
@@ -223,6 +224,50 @@ def test_lda_scaling():
                      'using covariance: %s' % solver)
 
 
+def test_lda_store_covariance():
+    # Test for slover 'lsqr' and 'eigen'
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+        assert_false(not hasattr(clf.fit(X, y), 'covariance_'))
+
+        # Test the deprecation
+        assert_warns(DeprecationWarning, clf.fit, X, y)
```

Comment:
Would be better to use `assert_warns_message` to clarify which deprecation this is.